src/Public/js/sis.selector.js

```diff
@@ -44,44 +43,31 @@ var Selector = function () {
       );
     },
     update = function (data) {
-      var liveStatus, s, time = parseInt(data["lastmod"]);
+      var liveStatus, s, studioNum, studioNumIndex, time = parseInt(data["lastmod"]);
       // Disregard data older than the latest update
       if (time <= lastTime) {
         return;
       }
-
       lastTime = parseInt(data["lastmod"]);
       // When called bet selectStudio, this isn't what I think it is
       // @todo, see if that can be bound nicer
       that.registerParam("selector-lasttime", lastTime);
 
       if (data["ready"]) {
-        if (!data["s1power"]) {
-          buttons[0].setAttribute("title", studios[0] + " Powered Off");
-          buttons[0].setAttribute("class", "selbtn poweredoff");
-          buttons[0].setAttribute("on", "false");
-        } else {
-          liveStatus = (data["studio"] == 1) ? "s1on" : "s1off";
-          buttons[0].setAttribute("title", studios[0]);
-          buttons[0].setAttribute("class", "selbtn poweredon " + liveStatus);
-          buttons[0].setAttribute("on", "true");
-        }
-
-        if (!data["s2power"]) {
-          buttons[1].setAttribute("title", studios[1] + " Powered Off");
-          buttons[1].setAttribute("class", "selbtn poweredoff");
-          buttons[1].setAttribute("on", "false");
-        } else {
-          liveStatus = (data["studio"] == 2) ? "s2on" : "s2off";
-          buttons[1].setAttribute("title", studios[1]);
-          buttons[1].setAttribute("class", "selbtn poweredon " + liveStatus);
-          buttons[1].setAttribute("on", "true");
+        for (studioNum = 1; studioNum <= 4; studioNum++) {
+          studioNumIndex = studioNum-1;
+          if (studioNum != 3 && data["s" + studioNum + "power"] == false) {
```

Comment:
invert this? having the false case first is a bit of a code smell. the `== false` makes me sad as well